libcxx/include/array

```diff
@@ -280,10 +280,11 @@ struct _LIBCPP_TEMPLATE_VIS array<_Tp, 0>
     typedef std::reverse_iterator<iterator>       reverse_iterator;
     typedef std::reverse_iterator<const_iterator> const_reverse_iterator;
 
-    typedef __conditional_t<is_const<_Tp>::value, const char, char> _CharType;
+    struct _EmptyAggregate { };
```

Comment:
We have `__empty` and I don't think we have to have this be specific to the instantiation, so maybe we want to use that instead?